### PR TITLE
LLM Assisted Tagging

### PR DESCRIPTION
**Pull request by sudan45**
- https://github.com/the-deep/server/issues/1544
## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/assisted_tagging/schema.py [hunk 1]
```diff
@@ -187,6 +203,12 @@ def resolve_prediction_tags(root, info, **kwargs):
     def resolve_geo_areas(root, info, **_):
         return info.context.dl.geo.draft_entry_geo_area.load(root.pk)
 
+    # TODO use dataloader
+    @staticmethod
+    def resolve_tags(root, info, **_):
+        return info.context.dl.assisted_tagging.llm_draft_entry_predications.load(root.pk)
+        # return LLMAssistedTaggingPredication.objects.get(draft_entry=root)
```
Comment (Rup-Narayan-Rajbanshi) [inline apps/assisted_tagging/schema.py]: Lets remove this commented line.

**Comment by Rup-Narayan-Rajbanshi** [comment]
Lets add dependent PR link in this PR. I think this PR has dependency with https://github.com/the-deep/server/pull/1541#pullrequestreview-2480895241. @sudan45 .

File: apps/project/serializers.py [hunk 2]
```diff
@@ -959,3 +959,21 @@ def create(self, validated_data):
...
+
+    def validate(self, data):
+        membership = ProjectMembership.objects.filter(
+            project=self.project,
+        )
+        if membership.count() == 1:
+            raise serializers.ValidationError('Cannot leave the project')
+        if membership.filter(role=ProjectRole.get_owner_role()).count() == 1:
+            raise serializers.ValidationError('Cannot leave the project')
+        return data
```
Comment (Rup-Narayan-Rajbanshi) [inline apps/project/serializers.py]: We need to update this as done in this PR:https://github.com/the-deep/server/pull/1541#pullrequestreview-2480895241 Or rebase this with that branch.

**Review by Rup-Narayan-Rajbanshi — CHANGES_REQUESTED** [review]

File: apps/deepl_integration/handlers.py [hunk 3]
```diff
@@ -1128,3 +1129,269 @@ def save_data(
...
+        common_attrs = dict(
+            model_version=model_version,
+            draft_entry_id=draft_entry.id,
+        )
+        LLMAssistedTaggingPredication.objects.create(
+            **common_attrs,
+            model_tags=tags
+        )
+        # draft_entry.prediction_status = DraftEntry.PredictionStatus.DONE
+        # draft_entry.save(update_fields='prediction_status')
```
Comment (Rup-Narayan-Rajbanshi) [inline apps/deepl_integration/handlers.py]: Remove this.

**Review by Rup-Narayan-Rajbanshi — CHANGES_REQUESTED** [review]

File: apps/deepl_integration/handlers.py (hunk 3 above)
Comment (Rup-Narayan-Rajbanshi) [reply apps/deepl_integration/handlers.py]: @sudan45

**Review by Rup-Narayan-Rajbanshi — APPROVED** [review]

**Review by thenav56 — APPROVED** [review]
All good other then the geo

**Review by susilnem — APPROVED** [review]
GTM!

**Review by susilnem — APPROVED** [review]
GTM!

**Pull request merged**